swarm.go

```diff
@@ -235,6 +262,26 @@ func (s *Swarm) addConn(tc transport.CapableConn, dir network.Direction) (*Conn,
 	s.notifyAll(func(f network.Notifiee) {
 		f.Connected(s, c)
 	})
+
+	// ---------------------------------------------------------
+	// Emit the Connectedness=Connected event on the bus if we just went
+	// from having no connections with the peer to having this connection with the peer.
+	indexForLk := len(p) - 1
+	stripe := &s.stripedPeerConnectivity[p[indexForLk]]
```

Comment:
p is a peerID string, and here we're taking the last byte. Is it possible for remote peerids to be biased so a sybil can make all its identities fall into the same stripe? would it make sense to use a hash function?

Reply:
Yes but this is really just best-effort in case the underlying event bus blocks. Honestly, I'm not really sure how critical it is. Thoughts @raulk?